core/ir/instr_api.h

```diff
@@ -738,6 +738,19 @@ DR_API
 uint
 instr_get_category(instr_t *instr);
 
+DR_API
+/**
+ * Fills \p category_names with a comma-separated string of categories.
```

Comment:
Hmm, it seems more flexible to instead provide a string name for an individual category so the user can combine them in whatever way they want, rather than hardcoding commas or a certain ordering?